verification/certifyKey_test.go

```diff
@@ -3,61 +3,521 @@
+	OidExtensionTcgDiceKpIdentityLoc.String(),
+	OidExtensionTcgDiceKpAttestLoc.String(),
+}
+
+// tcg-dice-Ueid OBJECT IDENTIFIER ::= {tcg-dice 4}
+//
+//	TcgUeid ::== SEQUENCE {
+//			ueid OCTET STRING
+//	}
+type TcgUeidExtension struct {
+	Ueid []uint8 `asn1:"ueid,implicit"`
+}
+
+// tcg-dice-MultiTcbInfo OBJECT IDENTIFIER ::= {tcg-dice 5}
+// DiceTcbInfoSeq ::= SEQUENCE SIZE (1..MAX) OF DiceTcbInfo
+//
+// tcg-dice-TcbInfo OBJECT IDENTIFIER ::= {tcg-dice 1}
+//
+// DiceTcbInfo 	::== SEQUENCE {
+// 		vendor		[0] IMPLICIT UTF8String OPTIONAL,
+// 		model 		[1] IMPLICIT UTF8String OPTIONAL,
+// 		version 	[2] IMPLICIT UTF8String OPTIONAL,
+// 		svn 		[3] IMPLICIT INTEGER OPTIONAL,
+// 		layer 		[4] IMPLICIT INTEGER OPTIONAL,
+// 		index 		[5] IMPLICIT INTEGER OPTIONAL,
+// 		fwids 		[6] IMPLICIT FWIDLIST OPTIONAL,
+// 		flags 		[7] IMPLICIT OperationalFlags OPTIONAL,
+//		vendorInfo 	[8] IMPLICIT OCTET STRING OPTIONAL,
+// 		type 		[9] IMPLICIT OCTET STRING OPTIONAL,
+// }
+//
+// FWIDLIST ::== SEQUENCE SIZE (1..MAX) OF FWID
+// FWID ::== SEQUENCE {
+// 		hashAlg 	OBJECT IDENTIFIER,
+// 		digest 		OCTET STRING
+// }
+//
+// OperationalFlags ::= BIT STRING {
+// 		notConfigured (0),
+// 		notSecure (1),
+// 		recovery (2),
+//  	debug (3)
+// }
+
+type Fwid struct {
+	HashAlg string
+	Digest  string
+}
+
+type DiceTcbInfo struct {
+	Fwids      []Fwid
+	VendorInfo string
+	Type       string
+}
+
+type TcgMultiTcbInfo struct {
+	DiceTcbInfos []DiceTcbInfo
+}
 
 func TestCertifyKey(t *testing.T) {
 
-	support_needed := []string{"AutoInit", "X509"}
-	instance, err := GetTestTarget(support_needed)
+	supportNeeded := []string{"AutoInit", "X509", "IsCA"}
+	instance, err := GetTestTarget(supportNeeded)
 	if err != nil {
 		if err.Error() == "Requested support is not supported in the emulator" {
-			t.Skipf("Warning: Failed executing TestCertifyKey command due to unsupported request. Hence, skipping the command execution")
+			t.Skipf("[WARNING]: Failed executing TestCertifyKey command due to unsupported request. Hence, skipping the command execution")
 		} else {
-			log.Fatal(err)
+			log.Fatalf("[FATAL]: %s", err.Error())
 		}
 	}
 	testCertifyKey(instance, t)
 }
 
 func TestCertifyKey_SimulationMode(t *testing.T) {
 
-	support_needed := []string{"AutoInit", "Simulation", "X509"}
-	instance, err := GetTestTarget(support_needed)
+	supportNeeded := []string{"AutoInit", "Simulation", "X509", "IsCA"}
+	instance, err := GetTestTarget(supportNeeded)
 	if err != nil {
 		if err.Error() == "Requested support is not supported in the emulator" {
-			t.Skipf("Warning: Failed executing TestCertifyKey_SimulationMode command due to unsupported request. Hence, skipping the command execution")
+			t.Skipf("[WARNING]: Failed executing TestCertifyKey_SimulationMode command due to unsupported request. Hence, skipping the command execution")
 		} else {
-			log.Fatal(err)
+			log.Fatalf("[FATAL]: %s", err.Error())
 		}
 	}
 	testCertifyKey(instance, t)
```

Comment:
This doesn't appear to actually be testing in simulation mode (it just tests that the target supports simulation mode). To test this, we should initializing a simulation context and calling CertifyKey on it.

Reply:
Addressed